### ISSUE TITLE
[BUG] Crash when deleting all nodes of a contour widget for smart contour

### ISSUE THREAD

**Issue by alabamagan** [wuzhuobin/blood_vessel#18]
InteractorStyleSmartContour2.cxx : 261

```
if (this->m_lastSlice != -1) {
    for (int i = 0; i < 2; ++i) {
        for (int j = 0; j < lists[i]->size(); j++) {
            vtkSmartPointer<vtkContourRepresentation> rep = lists[i]->at(j)->GetContourRepresentation();
            vtkSmartPointer<vtkPolyData> repPD = vtkSmartPointer<vtkPolyData>::New();
            rep->GetNodePolyData(repPD);
            (*polydataList[i])[this->m_lastSlice]->at(j)->DeepCopy(repPD); // Crash here
        }
    }
}
```

**Comment by alabamagan** [comment]
Also crash while redoing generate contour, at the same location, probably m_lastSlice is pointing to a deleted contour widget

**Comment by wuzhuobin** [comment]
fixed

**Comment by alabamagan** [comment]
fixed in 243aab3
